contracts/SafeMath.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity ^0.4.21;
+
+/**
+ * @title SafeMath
+ * @dev Math operations with safety checks that throw on error
+ */
+library SafeMath {
+
+  /**
+  * @dev Multiplies two numbers, throws on overflow.
+  */
+  function mul(uint256 a, uint256 b) internal pure returns (uint256) {
+    if (a == 0) {
+      return 0;
+    }
+    uint256 c = a * b;
+    assert(c / a == b);
+    return c;
+  }
+
+  /**
+  * @dev Integer division of two numbers, truncating the quotient.
+  */
+  function div(uint256 a, uint256 b) internal pure returns (uint256) {
+    // assert(b > 0); // Solidity automatically throws when dividing by 0
```

Comment:
We don't want a throw; here do we?
Assert/require/revert would probably be better

Reply:
Yes, Solidity will automatically throw/revert on division by zero.

Reply:
Does it revert or does it throw? Very different outcomes.

Reply:
From the solidity odcs:

An assert-style exception is generated in the following situations:
...
If you divide or modulo by zero (e.g. 5 / 0 or 23 % 0).

So, the OpenZeppelin removed the assert, because it is implicitly done by solidity already.

Reply:
I think it would be better to revert? Save users gas if you can?